lib/has_secure_password_argon2.rb

```diff
@@ -39,7 +39,7 @@ def authenticate(unencrypted_password)
       if password_digest.start_with?('$argon2')
         Argon2::Password.verify_password(unencrypted_password, password_digest, HasSecurePasswordArgon2.secret) && self
       elsif super(unencrypted_password)
-        self.password = hash_argon2_password(unencrypted_password)
+        self.password_digest = hash_argon2_password(unencrypted_password)
```

Comment:
This seems like the more sensible change to me
```suggestion
        self.password = unencrypted_password
```